stackslib/src/chainstate/nakamoto/coordinator/mod.rs

```diff
@@ -243,34 +238,151 @@ pub fn get_nakamoto_reward_cycle_info<U: RewardSetProvider>(
     let reward_cycle = burnchain
         .next_reward_cycle(burn_height)
         .expect("FATAL: no reward cycle for burn height");
-    let reward_start_height = burnchain.reward_cycle_to_block_height(reward_cycle);
 
     debug!("Processing reward set for Nakamoto reward cycle";
           "burn_height" => burn_height,
           "reward_cycle" => reward_cycle,
           "reward_cycle_length" => burnchain.pox_constants.reward_cycle_length,
           "prepare_phase_length" => burnchain.pox_constants.prepare_length);
 
+    let Some((rc_info, anchor_block_header)) = load_nakamoto_reward_set(
+        burn_height,
+        sortition_tip,
+        burnchain,
+        chain_state,
+        sort_db,
+        provider,
+    )?
+    else {
+        return Ok(None);
+    };
+
+    let block_id = match anchor_block_header.anchored_header {
+        StacksBlockHeaderTypes::Epoch2(..) => anchor_block_header.index_block_hash(),
+        StacksBlockHeaderTypes::Nakamoto(ref header) => header.block_id(),
+    };
+
+    info!(
+        "Anchor block selected";
+        "cycle" => reward_cycle,
+        "block_id" => %block_id,
+        "consensus_hash" => %anchor_block_header.consensus_hash,
+        "burn_height" => anchor_block_header.burn_header_height,
+        "anchor_chain_tip" => %anchor_block_header.index_block_hash(),
+        "anchor_chain_tip_height" => %anchor_block_header.burn_header_height,
+    );
+
+    return Ok(Some(rc_info));
+}
+
+/// Helper to get the Nakamoto reward set for a given reward cycle.
+/// In all but the first Nakamoto reward cycle, this will load up the stored reward set from the
+/// Nakamoto chain state.  In the first Nakamoto reward cycle, where the reward set is computed
+/// from epoch2 state, the reward set will be loaded from the sortition DB (which is the only place
+/// it will be stored).
```

Comment:
I want to push back on this a little bit -- is `load_nakamoto_reward_set()` even invoked for the first nakamoto reward cycle? That reward cycle starts during epoch 2.5, so the 2.x reward set calculation should be used for it. If this is the case, I think we should be more direct about how `load_nakamoto_reward_set()` is invoked (maybe erroring if its invoked at a bad height?) and also remove the fallback-to-2x logic.

Reply:
>  is load_nakamoto_reward_set() even invoked for the first nakamoto reward cycle?

Yes, it is -- by both the chains coordinator and by the miner.  This function takes a reward cycle in the Nakamoto epoch, and returns its reward set.  The first Nakamoto reward cycle's reward set happens to have been generated in the last reward cycle for epoch 2.5, which is why there's a special-case path for it.

I'd prefer to keep it this way, and have one place where the special-case code path is reached, than to duplicate it into the half-dozen other places where it would become necessary if it was removed from this function.

Reply:
Okay, that works for me.